File: LongestPalindromicSubstring.py
#This solution does not work for longer strings.
class Solution:
    def longestPalindrome(self, s: str) -> str:
        longestPalindromeString=""
        def checkPalindrome(x: str) -> bool:
            if len(x)==1:
                return True
            return x==x[::-1] 
        for i in range(1,len(s)+1):
            start=0
            while start+i<=len(s):
                if checkPalindrome(s[start:start+i]):
                    if len(s[start:start+i])>len(longestPalindromeString):
                        longestPalindromeString=s[start:start+i]
                start=start+1
        return longestPalindromeString

File: LongestPalindromicSubstring.java
class Solution {
    public String longestPalindrome(String s) {
        if(s==null || s.length()==0) return "";
        int start=0,end=0;
        for(int i=0;i<s.length();i++){
            int len=Math.max(palindromeCheck(s,i,i),palindromeCheck(s,i,i+1));
            if(len>end-start){
                start=i-(len-1)/2;
                end=i+len/2;
            }
        }
        return s.substring(start,end+1);
    }
    private int palindromeCheck(String s,int L,int R){
        while (L >= 0 && R < s.length() && s.charAt(L) == s.charAt(R)) {
            L--;
            R++;
        }
        return R - L - 1;
    }
}